### PR TITLE
33337: eliminated laminas-i18n and laminas-filter dependency

### LINKED ISSUE #33337: Eliminate laminas-i18n and laminas-filter dependency
laminas/laminas-i18n has a hidden dependency `laminas/laminas-filter` released its PHP 8 compatible version in January 2021, in order to accelerate future PHP compatibility work this dependency should be eliminated.

laminas/laminas-i18n and as a result laminas/laminas-filter dependencies should be eliminated.

Laminas\I18n\Filter\NumberParse is used in \Magento\Checkout\Model\Cart\RequestQuantityProcessor::process

### PR DESCRIPTION
**Pull request by korovitskyi**
### Description (*)

Eliminated laminas-i18n and laminas-filter dependency
Also it was created the private `filter` method to proceed with the native logic process.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#33337

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/557

### CODE REVIEW

**Comment by korovitskyi** [comment]
@magento run all tests

File: app/code/Magento/Checkout/Model/Cart/RequestQuantityProcessor.php [hunk 1]
```diff
@@ -76,4 +74,30 @@ public function prepareQuantity($quantity)
...
+     */
+    private function filter($quantity)
+    {
+        $formatter = new NumberFormatter($this->localeResolver->getLocale(), NumberFormatter::DEFAULT_STYLE);
+
+        if (is_array($quantity)) {
+            foreach ($quantity as $key => $qty) {
+                if (!is_float($qty) && !is_int($qty)) {
+                    $quantity[$key] = $formatter->parse($qty);
+                }
```
Comment (Den4ik) [inline app/code/Magento/Checkout/Model/Cart/RequestQuantityProcessor.php]: I believe that we can move this part into separate private method. In my opinion code will looks more clear

File: app/code/Magento/Checkout/Model/Cart/RequestQuantityProcessor.php [hunk 2]
```diff
@@ -76,4 +74,30 @@ public function prepareQuantity($quantity)
 
         return $quantity;
     }
+
+    /**
+     * Filter quantity value and parse it by region if needed.
+     *
+     * @param float|int|array $quantity
```
Comment (Den4ik) [inline app/code/Magento/Checkout/Model/Cart/RequestQuantityProcessor.php]: is `$quantity` could be the string?

File: composer.lock [hunk 3]
```diff
@@ -11345,5 +12034,6 @@
         "ext-zip": "*",
         "lib-libxml": "*"
     },
-    "platform-dev": []
+    "platform-dev": [],
+    "plugin-api-version": "1.1.0"
```
Comment (Den4ik) [inline composer.lock]: Please use composer v2

**Review by Den4ik — CHANGES_REQUESTED** [review]
@korovitskyi Thanks for your contribution.
Could you please check my comments.

Also if I'm correct we should create PR to `platform-health` branch. @sivaschenko, correct me if I'm wrong.

**Comment by korovitskyi** [comment]
@magento run all tests

**Comment by korovitskyi** [comment]
Hi, @Den4ik .
The changes were added.
Also was changed the PR branch to `platform-health`.
Thank you for the review.

**Comment by Den4ik** [comment]
@korovitskyi Please resolve merge conflict

**Comment by Den4ik** [comment]
All looks ok for me. I'll approve after fixing merge conflict and rerun tests

**Comment by korovitskyi** [comment]
@magento run all tests

**Comment by korovitskyi** [comment]
Hi, @Den4ik .
The conflicts have been reloved.

**Comment by Den4ik** [comment]
@magento run Functional Tests EE, Unit Tests

**Review by Den4ik — APPROVED** [review]
✅ Approved.
@korovitskyi Good job 👍

**Review by eduard13 — APPROVED** [review]

**Comment by andrewbess** [comment]
Hello @korovitskyi 
Could you please resolve merge-conflict.
Thank you in advance.

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@korovitskyi Please pay attention to https://github.com/magento/magento2/pull/33362#issuecomment-882863229

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
Closed due to opening new it https://github.com/magento/magento2/pull/33529

**Pull request closed without merging**